src/document/services/document.form.js

```diff
@@ -161,6 +163,11 @@ angular.module('engine.document')
                 }
             });
 
+            _.forEach(metricToRemove, function (metricId) {
```

Comment:
Why don't you do the removal around line 129 though? Removal from the other places is done there.

Reply:
...I don't know why :<